### ISSUE TITLE
High latency issue when using Predict API with text classification model with thousands of labels

### ISSUE THREAD

**Issue by spate141** [tensorflow/serving#853]
Hi,

I'm using TF serving to load text classification model which predicts 1000 labels. My prediction signature looks like this:
```
signature = predict_signature_def(
    inputs={'embedding': new_model.input}, 
    outputs={'dense': new_model.output} <-- predict 1000 labels
)
```

Issue is, when when I get the response back with following, getting  `response.outputs['dense'].float_val` is taking so much time.
```
response = stub.Predict(request_object, 10.0)
result = np.expand_dims(response.outputs['dense'].float_val, axis=0)
```

For example, when `request_object` contains 500 documents to classify, I get back the `response.outputs['dense'].float_val` of size `500,000` which takes most of the time and increase the latency!

Things that I already tried:
- Making batch of small size, so the model classify small batch of data and return back small set of response. This is still not optimal, as long as I'm receiving back all predictions values back for each batch, No matter how small batch I use ; it's always going to take certain time to receive the large response object when predicting thousands of labels. 

My previous Keras model provides function `model.predict(k=3)` which only returns `top-k` predictions. I was not able to figure out how to use functionality similar to this with serving. 

I would really appreciate If anyone can point me to correct part/documentation where I can use something similar to only get back `top-k` predictions with TensorFlow Serving. Or any suggestions on how to handle large size of predictions.

**EDIT:** I'm using similar setup like this, https://github.com/tensorflow/serving/issues/310#issuecomment-297015251

Thanks!

**Comment by gautamvasudevan** [comment]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow-serving

If you open a GitHub issue, it must be a bug, a feature request, or a
significant problem with documentation (for small docs fixes please send
a PR instead).

Thanks!
